ignite/contrib/handlers/time_profilers.py

```diff
@@ -437,3 +438,299 @@ def odict_to_str(d: Mapping):
+                h.__qualname__ if hasattr(h, "__qualname__") else h.__class__.__name__
+                for (h, _, _) in engine._event_handlers[e]
+                if not self._is_internal_handler(h)
+            ]
+            for e in engine._allowed_events
+        }
+
+        self._reset(self.event_handlers_names)
+
+        for e in engine._allowed_events:
+            for i, h in enumerate(engine._event_handlers[e]):
+                if not self._is_internal_handler(h):
+                    engine._event_handlers[e][i] = (self._timeit_handler_wrapper(h), (e,), {})
+
+        # processing timer
+        engine._event_handlers[Events.ITERATION_STARTED].append((self._processing_timer.reset, (), {}))
+        engine._event_handlers[Events.ITERATION_COMPLETED].insert(0, (self._timeit_processing, (), {}))
+
+        # dataflow timer
+        engine._event_handlers[Events.GET_BATCH_STARTED].append((self._dataflow_timer.reset, (), {}))
+        engine._event_handlers[Events.GET_BATCH_COMPLETED].insert(0, (self._timeit_dataflow, (), {}))
+
+        # revert back the wrapped handlers with original handlers at the end
+        engine._event_handlers[Events.COMPLETED].append((self._detach_profiler_handlers, (engine,), {}))
+
+    def attach(self, engine: Engine):
+        if not isinstance(engine, Engine):
+            raise TypeError("Argument engine should be ignite.engine.Engine, but given {}".format(type(engine)))
+
+        if not engine.has_event_handler(self._as_first_started):
+            engine._event_handlers[Events.STARTED].insert(0, (self._as_first_started, (engine,), {}))
+
+    def get_results(self):
+        """
+        Method to fetch the aggregated profiler results after the engine is run
+
+        .. code-block:: python
+
+            results = profiler.get_results()
+
+        """
+        total_eh_time = sum(
+            [
+                sum(self.event_handlers_times[e][h])
+                for e in self.event_handlers_times
+                for h in self.event_handlers_times[e]
+            ]
+        )
+        total_eh_time = round(float(total_eh_time), 5,)
+
+        def compute_basic_stats(data: Sequence):
+            # compute on non-zero data:
+            data = data[data > 0]
+            total = round(torch.sum(data).item(), 5) if len(data) > 0 else "not triggered"
+            min_index = ("None", "None")
+            max_index = ("None", "None")
+            mean = "None"
+            std = "None"
+            if len(data) > 1:
+                min_index = (round(torch.min(data).item(), 5), torch.argmin(data).item())
+                max_index = (round(torch.max(data).item(), 5), torch.argmax(data).item())
+                mean = round(torch.mean(data).item(), 5)
+                std = round(torch.std(data).item(), 5)
+            return [total, min_index, max_index, mean, std]
+
+        event_handler_stats = [
+            [h, e.name, *compute_basic_stats(torch.tensor(self.event_handlers_times[e][h]))]
```

Comment:
```suggestion
            [h, getattr(e, "name", str(e)), *compute_basic_stats(torch.tensor(self.event_handlers_times[e][h]))]
```
For custom events as `str` as they do not have `name` attribute.

Reply:
I thought python Enum types will always have name attribute. Is it not?

Reply:
oh got it. Didn't know we can use strings for registering custom events. 😃

Reply:
Yes, event can be `Events` or a str if registered.